backend/internal/flow/store/store.go

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package store provides the implementation for flow context persistence operations.
+package store
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/asgardeo/thunder/internal/flow/model"
+	"github.com/asgardeo/thunder/internal/system/database/provider"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+const loggerComponentName = "FlowStore"
+
+// StoreFlowContext stores the complete flow context in the database.
+func StoreFlowContext(ctx model.EngineContext) error {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName))
+
+	dbClient, err := provider.NewDBProvider().GetDBClient("runtime")
+	if err != nil {
+		logger.Error("Failed to get database client", log.Error(err))
+		return fmt.Errorf("failed to get database client: %w", err)
+	}
+	defer func() {
+		if closeErr := dbClient.Close(); closeErr != nil {
+			logger.Error("Failed to close database client", log.Error(closeErr))
+		}
+	}()
+
+	// Convert engine context to database model
+	dbModel, err := FromEngineContext(ctx)
+	if err != nil {
+		logger.Error("Failed to convert engine context to database model", log.Error(err))
+		return fmt.Errorf("failed to convert engine context to database model: %w", err)
+	}
+
+	logger.Debug("Storing flow context to database",
+		log.String("flowID", dbModel.FlowID),
+		log.String("currentNodeID", getStringValue(dbModel.CurrentNodeID)),
+		log.Bool("isAuthenticated", dbModel.IsAuthenticated),
+		log.String("userID", getStringValue(dbModel.UserID)))
+
+	// Store flow context
+	_, err = dbClient.Execute(QueryCreateFlowContext,
+		dbModel.FlowID, dbModel.AppID,
+		dbModel.CurrentNodeID, dbModel.CurrentActionID, dbModel.GraphID, dbModel.RuntimeData)
+	if err != nil {
+		logger.Error("Failed to create flow context", log.Error(err))
+		return fmt.Errorf("failed to create flow context: %w", err)
+	}
+
+	// Store flow user data
+	_, err = dbClient.Execute(QueryCreateFlowUserData,
+		dbModel.FlowID, dbModel.IsAuthenticated, dbModel.UserID,
+		dbModel.UserInputs, dbModel.UserAttributes)
+	if err != nil {
+		logger.Error("Failed to create flow user data", log.Error(err))
+		return fmt.Errorf("failed to create flow user data: %w", err)
+	}
```

Comment:
This should be a transaction. Refer https://github.com/asgardeo/thunder/blob/05706f63b70e6160be62a028b4f7e8c01fd523da/backend/internal/application/store/store.go#L60-L73